pkg/config/generate_test.go

```diff
@@ -26,39 +28,148 @@ properties:
         PublicSubdomain: ""
 `)
 
+func TestGenerate(t *testing.T) {
+	tests := map[string]struct {
+		omc *api.OpenShiftManagedCluster
+	}{
+		"test generate new": {
+			omc: fixtures.NewTestOpenShiftCluster(),
+		},
+		//TODO "test generate doesn't overwrite": {},
+	}
+
+	for name, test := range tests {
+		err := Generate(test.omc)
+		if err != nil {
+			t.Errorf("%s received generation error %v", name, err)
+			continue
+		}
+		testRequiredFields(test.omc, t)
+		// check mutation
+	}
+}
+
+func testRequiredFields(omc *api.OpenShiftManagedCluster, t *testing.T) {
+	assert := func(c bool, name string) {
+		if c {
```

Comment:
sorry, I'd specifically do !c here and reverse all the booleans in the callers, so that they read "assert(thing_that_should_be_true)" not "assert(thing_that_should_be_false)"